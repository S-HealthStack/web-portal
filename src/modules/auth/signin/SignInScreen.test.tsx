import React from 'react';
import 'jest-styled-components';
import '@testing-library/jest-dom/extend-expect';
import { act, render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from 'styled-components';
import { Provider } from 'react-redux';
import { ConnectedRouter, getLocation } from 'connected-react-router';
import theme from 'src/styles/theme';
import { AppDispatch, store } from 'src/modules/store/store';
import { history, Path } from 'src/modules/navigation/store';
import SignInScreen from 'src/modules/auth/signin/SignInScreen';
import { authSlice, isAuthorizedSelector } from 'src/modules/auth/auth.slice';
import { matchPath } from 'react-router-dom';
import { STORAGE_REFRESH_TOKEN_KEY, STORAGE_TOKEN_KEY } from 'src/modules/auth/utils';

// eslint-disable-next-line prefer-destructuring
const dispatch: AppDispatch = store.dispatch;

const passwordValue = 'samsung';

describe('SignInScreen', () => {
  beforeEach(() => {
    dispatch(authSlice.actions.clearAuth());
    localStorage.removeItem(STORAGE_TOKEN_KEY);
    localStorage.removeItem(STORAGE_REFRESH_TOKEN_KEY);
  });

  it('should sign in', async () => {
    await act(() => {
      render(
        <ThemeProvider theme={theme}>
          <Provider store={store}>
            <ConnectedRouter history={history}>
              <SignInScreen />
            </ConnectedRouter>
          </Provider>
        </ThemeProvider>
      );
    });

    expect(await screen.findByTestId('signin-screen')).toMatchSnapshot();

    const email = await screen.findByTestId('signin-screen-email');
    const password = await screen.findByTestId('signin-screen-password');
    const remember = await screen.findByTestId('signin-screen-remember');
    const send = await screen.findByTestId('signin-screen-send');

    const emailValue = 'hello@samsung.com';
    await userEvent.type(email, emailValue);
    expect(email).toHaveValue(emailValue);

    await userEvent.type(password, passwordValue);
    expect(password).toHaveValue(passwordValue);

    await userEvent.click(remember);
    expect(remember).toBeChecked();

    await userEvent.click(send);

    await waitFor(() => expect(isAuthorizedSelector(store.getState())).toBeTruthy());

    expect(
      matchPath(getLocation(store.getState()).pathname, {
        path: Path.Overview,
        exact: true,
      })
    ).not.toBeNull();
  });

  it('[NEGATIVE] should sign in with incorrect email', async () => {
    await act(() => {
      render(
        <ThemeProvider theme={theme}>
          <Provider store={store}>
            <ConnectedRouter history={history}>
              <SignInScreen />
            </ConnectedRouter>
          </Provider>
        </ThemeProvider>
      );
    });

    expect(await screen.findByTestId('signin-screen')).toMatchSnapshot();

    const email = await screen.findByTestId('signin-screen-email');
    const password = await screen.findByTestId('signin-screen-password');
    const remember = await screen.findByTestId('signin-screen-remember');
    const send = await screen.findByTestId('signin-screen-send');
    const error = await screen.findByTestId('signin-screen-error');

    const unsatisfactoryEmailValue = 'hello@example.com';
    await userEvent.type(email, unsatisfactoryEmailValue);
    expect(email).toHaveValue(unsatisfactoryEmailValue);

    await userEvent.type(password, passwordValue);
    expect(password).toHaveValue(passwordValue);

    await userEvent.click(remember);
    expect(remember).toBeChecked();

    await userEvent.click(send);

    await waitFor(() => expect(isAuthorizedSelector(store.getState())).toBeFalsy());

    expect(error).toHaveTextContent('Incorrect email or password, please try again.');
  });
});
