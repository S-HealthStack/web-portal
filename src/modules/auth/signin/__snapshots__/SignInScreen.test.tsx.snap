// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInScreen [NEGATIVE] should sign in with incorrect email 1`] = `
.c29 {
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: default;
  -webkit-user-drag: none;
}

.c29:hover {
  cursor: pointer;
}

.c25 {
  position: absolute;
  will-change: opacity,transform;
  border-radius: 50%;
  -webkit-animation: bykygw 600ms cubic-bezier(0.4,0,0.2,1) forwards;
  animation: bykygw 600ms cubic-bezier(0.4,0,0.2,1) forwards;
  z-index: 0;
  background-color: #E3EAFB;
  pointer-events: none;
}

.c27 {
  position: absolute;
  will-change: opacity,transform;
  border-radius: 50%;
  -webkit-animation: bykygw 600ms cubic-bezier(0.4,0,0.2,1) forwards;
  animation: bykygw 600ms cubic-bezier(0.4,0,0.2,1) forwards;
  z-index: 0;
  pointer-events: none;
}

.c22 {
  position: relative;
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 0.25rem;
  width: 7.5rem;
  white-space: nowrap;
  padding: 0;
  overflow: hidden;
  -webkit-transition-duration: 0;
  transition-duration: 0;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  height: 2.5rem;
  border: none;
  background-color: transparent;
}

.c22 > div {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 600;
  font-style: normal;
}

.c22 > div svg {
  margin-right: 0.25rem;
}

.c22 > div {
  color: #4475E3;
}

.c22 > div svg {
  fill: #4475E3;
}

.c22:hover:enabled > div {
  color: #2F65E0;
}

.c22:hover:enabled > div svg {
  fill: #2F65E0;
}

.c22:hover:enabled:active > div {
  color: #2057D5;
}

.c22:hover:enabled:active > div svg {
  fill: #2057D5;
}

.c22:disabled > div {
  color: #B3C6F1;
}

.c22:disabled > div svg {
  fill: #B3C6F1;
}

.c22:focus-visible {
  outline: 0.125rem solid #C7D6F7;
}

.c22:focus-visible > div {
  color: #4475E3;
}

.c22:focus-visible > div svg {
  fill: #4475E3;
}

.c26 {
  position: relative;
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 0.25rem;
  width: 100%;
  white-space: nowrap;
  padding: 0;
  overflow: hidden;
  -webkit-transition-duration: 400ms;
  transition-duration: 400ms;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  height: 3rem;
  border: none;
  background-color: #4475E3;
}

.c26 > div {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 600;
  font-style: normal;
}

.c26 > div svg {
  margin-right: 0.25rem;
}

.c26 > div {
  color: #FFFFFF;
}

.c26 > div svg {
  fill: #FFFFFF;
}

.c26:hover:enabled {
  background-color: #2F65E0;
}

.c26:disabled {
  background-color: #B3C6F1;
}

.c26:focus-visible:enabled {
  border: 0.125rem solid #FFFFFF;
  background-color: #4475E3;
  outline: 0.125rem solid #C7D6F7;
}

.c24 {
  position: relative;
  z-index: 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transform: translate3d(0,0,0.0625rem);
  -ms-transform: translate3d(0,0,0.0625rem);
  transform: translate3d(0,0,0.0625rem);
}

.c18 {
  display: inline-grid;
  gap: 0.25rem;
  grid-template-columns: 2.5rem 1fr;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c20 {
  width: 2.5rem;
  height: 2.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c21 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #474747;
  margin-top: 0.59375rem;
  -webkit-transition: color 300ms cubic-bezier(0.45,0.05,0,1);
  transition: color 300ms cubic-bezier(0.45,0.05,0,1);
}

.c16 {
  display: inline-block;
}

.c16 input {
  display: none;
}

.c16 .c19 > svg > :nth-child(1) {
  -webkit-transition: stroke 300ms cubic-bezier(0.45,0.05,0,1),fill 300ms cubic-bezier(0.45,0.05,0,1);
  transition: stroke 300ms cubic-bezier(0.45,0.05,0,1),fill 300ms cubic-bezier(0.45,0.05,0,1);
  stroke: #B3C6F1;
  fill: #FFFFFF;
}

.c16:hover > .c17 {
  cursor: pointer;
}

.c16:hover .c19 > svg > :nth-child(1) {
  fill: #ECF1FC;
  stroke: #B3C6F1;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.5rem;
}

.c10 > svg {
  position: relative;
  bottom: 2.6875rem;
  left: calc(100% - 0.5rem);
}

.c12 > svg {
  position: relative;
  bottom: 2.6875rem;
  left: calc(100% - 2rem);
}

.c9 {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 400;
  font-style: normal;
  color: #474747;
  box-sizing: border-box;
  height: 3.5rem;
  width: 100%;
  margin: 0;
  padding: 1rem;
  background-color: #F7F8FA;
  border: 0.0625rem solid #F7F8FA;
  border-radius: 0.25rem;
  -webkit-transition: border 300ms cubic-bezier(0.45,0.05,0,1);
  transition: border 300ms cubic-bezier(0.45,0.05,0,1);
  caret-color: #4475E3;
}

.c9:hover:enabled {
  border-color: #2F65E0;
}

.c9:active:enabled,
.c9:focus-visible {
  outline: none;
  border-color: #4475E3;
}

.c9:disabled {
  color: #DBDBDB;
  border-color: rgba(219,219,219,0.2);
  background-color: rgba(219,219,219,0.2);
}

.c9::-webkit-input-placeholder {
  color: #808080;
}

.c9::-moz-placeholder {
  color: #808080;
}

.c9:-ms-input-placeholder {
  color: #808080;
}

.c9::placeholder {
  color: #808080;
}

.c9:disabled::-webkit-input-placeholder {
  color: #DBDBDB;
}

.c9:disabled::-moz-placeholder {
  color: #DBDBDB;
}

.c9:disabled:-ms-input-placeholder {
  color: #DBDBDB;
}

.c9:disabled::placeholder {
  color: #DBDBDB;
}

.c9:-webkit-autofill,
.c9:-webkit-autofill:hover,
.c9:-webkit-autofill:focus {
  border: 0.0625rem solid #4475E3;
  -webkit-transition: background-color 5000s ease-in-out 0s;
  transition: background-color 5000s ease-in-out 0s;
  background-color: #F7F8FA !important;
  background-image: unset !important;
  -webkit-text-fill-color: #838383;
}

.c9:-webkit-autofill:focus {
  border-color: #4475E3;
}

.c7 {
  max-height: 3.5rem;
}

.c7:hover .c8:enabled {
  border-color: #2F65E0;
}

.c6 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #474747;
  gap: 0.5rem;
  height: 1.125rem;
}

.c11 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #D14343;
  padding-left: 1rem;
  height: 1.0625rem;
}

.c13 {
  background: none;
  border: none;
  padding: 0;
  height: 1.5rem;
  width: 1.5rem;
}

.c13 svg {
  fill: #4475E3;
}

.c13:active,
.c13:enabled:hover {
  cursor: pointer;
}

.c13:active svg,
.c13:enabled:hover svg {
  fill: #4475E3;
}

.c13:disabled svg {
  fill: #DBDBDB;
}

.c14 {
  position: relative;
  bottom: 2.6875rem;
  left: calc(100% - 2rem);
}

.c14:hover {
  cursor: pointer;
}

.c1 {
  color: #474747;
  min-width: 42.6875rem;
  width: 55.556vw;
  min-height: 32.015625rem;
  height: 41.667vw;
  max-height: 90vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: auto;
  background-color: #FFFFFF;
  box-shadow: 0 0 0.125rem 0rem rgba(0,0,0,0.15);
  border-radius: 0.25rem;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c3 {
  font-family: Inter,sans-serif;
  font-size: 2rem;
  line-height: 2.625rem;
  font-weight: 600;
  font-style: normal;
  margin: 1rem auto 0.5rem;
  text-align: center;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
}

.c2 {
  width: 28rem;
  margin: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  margin: 2.5rem auto 2.625rem;
}

.c15 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  height: 3.9375rem;
  color: #4475E3;
}

.c23 {
  height: 2.125rem;
}

.c23:disabled > div {
  color: #838383;
}

.c30 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #D14343;
  height: 2.5rem;
  width: 100%;
  text-align: center;
  padding-top: 1.25rem;
}

.c28 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #474747;
  margin-top: 0.5rem;
}

.c28 a {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 600;
  font-style: normal;
  color: #4475E3;
  display: inline-block;
  margin-left: 0.5rem;
}

@media (max-height:31.9375rem) {
  .c0 {
    overflow-y: scroll;
  }
}

@media (max-width:42.625rem) {
  .c0 {
    overflow-x: scroll;
  }
}

<div
  class="c0"
  data-testid="signin-screen"
>
  <div
    class="c1"
    width="55.556"
  >
    <div
      class="c2"
    >
      <div
        class="c3 c4"
      >
        Sign in
      </div>
      <div
        class="c5"
      >
        <div
          class="c6"
          data-testid="input-description"
        >
          Email
        </div>
        <div
          class="c7"
        >
          <input
            aria-label="Email"
            class="c8 c9"
            data-testid="signin-screen-email"
            name="email"
            type="email"
            value=""
          />
          <div
            class="c10"
          />
        </div>
        <div
          class="c11"
          data-testid="input-error"
        >
           
        </div>
      </div>
      <div
        class="c5"
      >
        <div
          class="c6"
          data-testid="input-description"
        >
          Password
        </div>
        <div
          class="c7"
        >
          <input
            aria-label="Password"
            autocomplete="new-password"
            class="c8 c9"
            data-testid="signin-screen-password"
            name="password"
            type="password"
            value=""
          />
          <div
            class="c12"
          >
            <button
              aria-label="Show Password"
              class="c13 c14"
              color="primary"
            >
              <svg
                data-testid="icon"
              />
            </button>
          </div>
        </div>
        <div
          class="c11"
          data-testid="input-error"
        >
           
        </div>
      </div>
      <div
        class="c15"
      >
        <div
          class="c16"
        >
          <label
            class="c17 c18"
            for="checkbox_1"
          >
            <div
              class="c19 c20"
            >
              <svg />
            </div>
            <div
              class="c21"
              data-testid="checkbox-label"
            >
              <div>
                Remember me
              </div>
            </div>
          </label>
          <input
            data-testid="signin-screen-remember"
            id="checkbox_1"
            type="checkbox"
          />
        </div>
        <button
          class="c22 c23"
          disabled=""
          fill="text"
          style="overflow: hidden; position: relative;"
          width="120"
        >
          <div
            class="c24"
            data-testid="content"
          >
            Forgot Password?
          </div>
          <div
            class="c25"
            color="primaryLightPressed"
            opacity="1,1"
            scale="0"
            style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
          />
        </button>
      </div>
      <button
        class="c26"
        data-testid="signin-screen-send"
        disabled=""
        fill="solid"
        style="overflow: hidden; position: relative;"
      >
        <div
          class="c24"
          data-testid="content"
        >
          Sign In
        </div>
        <div
          class="c27"
          color="primaryPressed"
          opacity="1,1"
          scale="0"
          style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
        />
      </button>
      <div
        class="c28"
      >
        Don’t have an account?
        <a
          class="c29"
          href="/account/create"
        >
          Create an account
        </a>
      </div>
      <div
        class="c30"
        data-testid="signin-screen-error"
      />
    </div>
  </div>
</div>
`;

exports[`SignInScreen should sign in 1`] = `
.c29 {
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: default;
  -webkit-user-drag: none;
}

.c29:hover {
  cursor: pointer;
}

.c25 {
  position: absolute;
  will-change: opacity,transform;
  border-radius: 50%;
  -webkit-animation: bykygw 600ms cubic-bezier(0.4,0,0.2,1) forwards;
  animation: bykygw 600ms cubic-bezier(0.4,0,0.2,1) forwards;
  z-index: 0;
  background-color: #E3EAFB;
  pointer-events: none;
}

.c27 {
  position: absolute;
  will-change: opacity,transform;
  border-radius: 50%;
  -webkit-animation: bykygw 600ms cubic-bezier(0.4,0,0.2,1) forwards;
  animation: bykygw 600ms cubic-bezier(0.4,0,0.2,1) forwards;
  z-index: 0;
  pointer-events: none;
}

.c22 {
  position: relative;
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 0.25rem;
  width: 7.5rem;
  white-space: nowrap;
  padding: 0;
  overflow: hidden;
  -webkit-transition-duration: 0;
  transition-duration: 0;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  height: 2.5rem;
  border: none;
  background-color: transparent;
}

.c22 > div {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 600;
  font-style: normal;
}

.c22 > div svg {
  margin-right: 0.25rem;
}

.c22 > div {
  color: #4475E3;
}

.c22 > div svg {
  fill: #4475E3;
}

.c22:hover:enabled > div {
  color: #2F65E0;
}

.c22:hover:enabled > div svg {
  fill: #2F65E0;
}

.c22:hover:enabled:active > div {
  color: #2057D5;
}

.c22:hover:enabled:active > div svg {
  fill: #2057D5;
}

.c22:disabled > div {
  color: #B3C6F1;
}

.c22:disabled > div svg {
  fill: #B3C6F1;
}

.c22:focus-visible {
  outline: 0.125rem solid #C7D6F7;
}

.c22:focus-visible > div {
  color: #4475E3;
}

.c22:focus-visible > div svg {
  fill: #4475E3;
}

.c26 {
  position: relative;
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 0.25rem;
  width: 100%;
  white-space: nowrap;
  padding: 0;
  overflow: hidden;
  -webkit-transition-duration: 400ms;
  transition-duration: 400ms;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  height: 3rem;
  border: none;
  background-color: #4475E3;
}

.c26 > div {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 600;
  font-style: normal;
}

.c26 > div svg {
  margin-right: 0.25rem;
}

.c26 > div {
  color: #FFFFFF;
}

.c26 > div svg {
  fill: #FFFFFF;
}

.c26:hover:enabled {
  background-color: #2F65E0;
}

.c26:disabled {
  background-color: #B3C6F1;
}

.c26:focus-visible:enabled {
  border: 0.125rem solid #FFFFFF;
  background-color: #4475E3;
  outline: 0.125rem solid #C7D6F7;
}

.c24 {
  position: relative;
  z-index: 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transform: translate3d(0,0,0.0625rem);
  -ms-transform: translate3d(0,0,0.0625rem);
  transform: translate3d(0,0,0.0625rem);
}

.c18 {
  display: inline-grid;
  gap: 0.25rem;
  grid-template-columns: 2.5rem 1fr;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c20 {
  width: 2.5rem;
  height: 2.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c21 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #474747;
  margin-top: 0.59375rem;
  -webkit-transition: color 300ms cubic-bezier(0.45,0.05,0,1);
  transition: color 300ms cubic-bezier(0.45,0.05,0,1);
}

.c16 {
  display: inline-block;
}

.c16 input {
  display: none;
}

.c16 .c19 > svg > :nth-child(1) {
  -webkit-transition: stroke 300ms cubic-bezier(0.45,0.05,0,1),fill 300ms cubic-bezier(0.45,0.05,0,1);
  transition: stroke 300ms cubic-bezier(0.45,0.05,0,1),fill 300ms cubic-bezier(0.45,0.05,0,1);
  stroke: #B3C6F1;
  fill: #FFFFFF;
}

.c16:hover > .c17 {
  cursor: pointer;
}

.c16:hover .c19 > svg > :nth-child(1) {
  fill: #ECF1FC;
  stroke: #B3C6F1;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.5rem;
}

.c10 > svg {
  position: relative;
  bottom: 2.6875rem;
  left: calc(100% - 0.5rem);
}

.c12 > svg {
  position: relative;
  bottom: 2.6875rem;
  left: calc(100% - 2rem);
}

.c9 {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 400;
  font-style: normal;
  color: #474747;
  box-sizing: border-box;
  height: 3.5rem;
  width: 100%;
  margin: 0;
  padding: 1rem;
  background-color: #F7F8FA;
  border: 0.0625rem solid #F7F8FA;
  border-radius: 0.25rem;
  -webkit-transition: border 300ms cubic-bezier(0.45,0.05,0,1);
  transition: border 300ms cubic-bezier(0.45,0.05,0,1);
  caret-color: #4475E3;
}

.c9:hover:enabled {
  border-color: #2F65E0;
}

.c9:active:enabled,
.c9:focus-visible {
  outline: none;
  border-color: #4475E3;
}

.c9:disabled {
  color: #DBDBDB;
  border-color: rgba(219,219,219,0.2);
  background-color: rgba(219,219,219,0.2);
}

.c9::-webkit-input-placeholder {
  color: #808080;
}

.c9::-moz-placeholder {
  color: #808080;
}

.c9:-ms-input-placeholder {
  color: #808080;
}

.c9::placeholder {
  color: #808080;
}

.c9:disabled::-webkit-input-placeholder {
  color: #DBDBDB;
}

.c9:disabled::-moz-placeholder {
  color: #DBDBDB;
}

.c9:disabled:-ms-input-placeholder {
  color: #DBDBDB;
}

.c9:disabled::placeholder {
  color: #DBDBDB;
}

.c9:-webkit-autofill,
.c9:-webkit-autofill:hover,
.c9:-webkit-autofill:focus {
  border: 0.0625rem solid #4475E3;
  -webkit-transition: background-color 5000s ease-in-out 0s;
  transition: background-color 5000s ease-in-out 0s;
  background-color: #F7F8FA !important;
  background-image: unset !important;
  -webkit-text-fill-color: #838383;
}

.c9:-webkit-autofill:focus {
  border-color: #4475E3;
}

.c7 {
  max-height: 3.5rem;
}

.c7:hover .c8:enabled {
  border-color: #2F65E0;
}

.c6 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #474747;
  gap: 0.5rem;
  height: 1.125rem;
}

.c11 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #D14343;
  padding-left: 1rem;
  height: 1.0625rem;
}

.c13 {
  background: none;
  border: none;
  padding: 0;
  height: 1.5rem;
  width: 1.5rem;
}

.c13 svg {
  fill: #4475E3;
}

.c13:active,
.c13:enabled:hover {
  cursor: pointer;
}

.c13:active svg,
.c13:enabled:hover svg {
  fill: #4475E3;
}

.c13:disabled svg {
  fill: #DBDBDB;
}

.c14 {
  position: relative;
  bottom: 2.6875rem;
  left: calc(100% - 2rem);
}

.c14:hover {
  cursor: pointer;
}

.c1 {
  color: #474747;
  min-width: 42.6875rem;
  width: 55.556vw;
  min-height: 32.015625rem;
  height: 41.667vw;
  max-height: 90vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: auto;
  background-color: #FFFFFF;
  box-shadow: 0 0 0.125rem 0rem rgba(0,0,0,0.15);
  border-radius: 0.25rem;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c3 {
  font-family: Inter,sans-serif;
  font-size: 2rem;
  line-height: 2.625rem;
  font-weight: 600;
  font-style: normal;
  margin: 1rem auto 0.5rem;
  text-align: center;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
}

.c2 {
  width: 28rem;
  margin: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  margin: 2.5rem auto 2.625rem;
}

.c15 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  height: 3.9375rem;
  color: #4475E3;
}

.c23 {
  height: 2.125rem;
}

.c23:disabled > div {
  color: #838383;
}

.c30 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #D14343;
  height: 2.5rem;
  width: 100%;
  text-align: center;
  padding-top: 1.25rem;
}

.c28 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #474747;
  margin-top: 0.5rem;
}

.c28 a {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 600;
  font-style: normal;
  color: #4475E3;
  display: inline-block;
  margin-left: 0.5rem;
}

@media (max-height:31.9375rem) {
  .c0 {
    overflow-y: scroll;
  }
}

@media (max-width:42.625rem) {
  .c0 {
    overflow-x: scroll;
  }
}

<div
  class="c0"
  data-testid="signin-screen"
>
  <div
    class="c1"
    width="55.556"
  >
    <div
      class="c2"
    >
      <div
        class="c3 c4"
      >
        Sign in
      </div>
      <div
        class="c5"
      >
        <div
          class="c6"
          data-testid="input-description"
        >
          Email
        </div>
        <div
          class="c7"
        >
          <input
            aria-label="Email"
            class="c8 c9"
            data-testid="signin-screen-email"
            name="email"
            type="email"
            value=""
          />
          <div
            class="c10"
          />
        </div>
        <div
          class="c11"
          data-testid="input-error"
        >
           
        </div>
      </div>
      <div
        class="c5"
      >
        <div
          class="c6"
          data-testid="input-description"
        >
          Password
        </div>
        <div
          class="c7"
        >
          <input
            aria-label="Password"
            autocomplete="new-password"
            class="c8 c9"
            data-testid="signin-screen-password"
            name="password"
            type="password"
            value=""
          />
          <div
            class="c12"
          >
            <button
              aria-label="Show Password"
              class="c13 c14"
              color="primary"
            >
              <svg
                data-testid="icon"
              />
            </button>
          </div>
        </div>
        <div
          class="c11"
          data-testid="input-error"
        >
           
        </div>
      </div>
      <div
        class="c15"
      >
        <div
          class="c16"
        >
          <label
            class="c17 c18"
            for="checkbox_1"
          >
            <div
              class="c19 c20"
            >
              <svg />
            </div>
            <div
              class="c21"
              data-testid="checkbox-label"
            >
              <div>
                Remember me
              </div>
            </div>
          </label>
          <input
            data-testid="signin-screen-remember"
            id="checkbox_1"
            type="checkbox"
          />
        </div>
        <button
          class="c22 c23"
          disabled=""
          fill="text"
          style="overflow: hidden; position: relative;"
          width="120"
        >
          <div
            class="c24"
            data-testid="content"
          >
            Forgot Password?
          </div>
          <div
            class="c25"
            color="primaryLightPressed"
            opacity="1,1"
            scale="0"
            style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
          />
        </button>
      </div>
      <button
        class="c26"
        data-testid="signin-screen-send"
        disabled=""
        fill="solid"
        style="overflow: hidden; position: relative;"
      >
        <div
          class="c24"
          data-testid="content"
        >
          Sign In
        </div>
        <div
          class="c27"
          color="primaryPressed"
          opacity="1,1"
          scale="0"
          style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
        />
      </button>
      <div
        class="c28"
      >
        Don’t have an account?
        <a
          class="c29"
          href="/account/create"
        >
          Create an account
        </a>
      </div>
      <div
        class="c30"
        data-testid="signin-screen-error"
      />
    </div>
  </div>
</div>
`;
